op-gateway/src/OPProofService.ts

```diff
@@ -5,72 +5,99 @@ import {
 } from '@ensdomains/evm-gateway';
 import { type JsonRpcBlock } from '@ethereumjs/block';
 import { AbiCoder, Contract, JsonRpcProvider, type AddressLike } from 'ethers';
+import OPOutputLookupABI from './OPOutputLookup.js'
+
+export class InvalidOptimismPortalError extends Error {
+  constructor() {
+    super('OptimismPortal is invalid');
+  }
+}
+
+export class DisputeGameChallengedError extends Error {
+  constructor() {
+    super('Dispute game is challenged');
+  }
+}
+
+export enum OPWitnessProofType {
+  L2OutputOracle = 0,
+  DisputeGame = 1,
+}
 
 export interface OPProvableBlock {
   number: number;
-  l2OutputIndex: number;
+  proofType: OPWitnessProofType;
+  index: number;
 }
 
-const L2_OUTPUT_ORACLE_ABI = [
-  'function latestOutputIndex() external view returns (uint256)',
-  'function getL2Output(uint256 _l2OutputIndex) external view returns (tuple(bytes32 outputRoot, uint128 timestamp, uint128 l2BlockNumber))',
-];
+// OPOutputLookup contract is deployed with deterministic deployment
```

Comment:
Can you link to the code/repo here?

Reply:
https://github.com/Opti-domains/evmgateway/blob/op-fault-proof-new/op-verifier/contracts/lib/OPOutputLookup.sol